tools/pythonpkg/src/python_replacement_scan.cpp

```diff
@@ -16,32 +19,65 @@ namespace duckdb {
 
 static void CreateArrowScan(const string &name, py::object entry, TableFunctionRef &table_function,
                             vector<unique_ptr<ParsedExpression>> &children, ClientProperties &client_properties,
-                            PyArrowObjectType type, DBConfig &config) {
-
-	if (type == PyArrowObjectType::PyCapsuleInterface) {
-		entry = entry.attr("__arrow_c_stream__")();
-		type = PyArrowObjectType::PyCapsule;
-	}
+                            PyArrowObjectType type, DBConfig &config, DatabaseInstance &db) {
+	py::gil_scoped_acquire acquire;
```

Comment:
This sounds like there is a path that leads here that doesn't hold the gil, which is a bug further up.
I don't think this is the right fix, the gil should probably be acquired further up, as all the other paths that lead here will already hold the gil.